enhancements/node-tuning/containerize-tuned.md

```diff
@@ -0,0 +1,255 @@
+  ExecStop=/usr/bin/podman stop -t 20 openshift-tuned
+  ExecStopPost=/usr/bin/podman rm -f --ignore openshift-tuned
+
+```
+
+This approach would work even for deployments such as HyperShift, openshift-ansible and likely even IBM ROKS; this still wouldn't work for custom deployments with RHEL.
+
+* Prior to the tuned main process, all necessary tuned directories and known profiles will be created and placed on the node.
+
+* This containerized approach requires having a path to custom tuned profiles directory configurable or at least in subdirectory /etc/tuned/profiles, in order to mount them as a whole in containerized environments. 
+This will have to be implemented on [Tuned](https://github.com/redhat-performance/tuned) - [RHEL-26157](https://issues.redhat.com/browse/RHEL-26157)
+
+* Allow static key IPIs to be disabled (caused when sockets are opened with timestamping enabled). Currently these IPIs cause latency hits for low latency applications. This applies to RHEL and also to OpenShift deployments. Currently, to prevent these IPIs, tuned has an [rtentsk plugin](https://github.com/redhat-performance/tuned/blob/master/tuned/plugins/plugin_rtentsk.py) that opens a socket with timestamping enabled and holds this socket open.
+This will have to be implemented on RHEL + RHCOS - [RHEL-23928](https://issues.redhat.com/browse/RHEL-23928)
+
+### Workflow Description
+
+N/A
+
+### API Extensions
+
+No API changes
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+TBD
+
+#### Standalone Clusters
+
+#### Single-node Deployments or MicroShift
+
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+TBD
+
+### Implementation Details/Notes/Constraints
+
+* “Accumulating” NTO images during upgrades.
+* Even with “podmanized” TuneD, we’ll likely not be able to resolve issues such as [these](https://github.com/openshift/machine-config-operator/pull/2944#issuecomment-1029739561) and centralized the tuning into NTO – chicken-and-egg with default openshift TuneD profiles. Rendering them via MCO/installer, doing some built-in NTO bootstrap or using host-native TuneD would help.
+* We need to ensure TuneD is not blocking Kubelet system service while starting before it. 15s of TuneD starting would be an issue. 
+
+### Risks and Mitigations
+
+* The [pod label based matching](https://docs.openshift.com/container-platform/4.15/scalability_and_performance/using-node-tuning-operator.html#accessing-an-example-node-tuning-operator-specification_node-tuning-operator) must stay working as it was never officially deprecated although the official warning exists.
+  * Disabling that functionality when PerformanceProfile is in place should be considered as a viable option.
+
+* Extra reboot during can be caused from the machine config for installation/upgrade.
+  * This is potentially solvable by providing a MC manifest for the TuneD systemd service at bootstrap.
```

Comment:
But the tuned profile wouldn't be available during bootstrap - so how does the initial application of the tuned profile work in that case (when it has been created by NTO)?

Reply:
There was some preliminary attempt to do this at bootstrap but I wasn't part of that effort. Yanir/Martin can probably comment.  Not only the TuneD profiles, but also the NTO/TuneD image needs to be identified.  At bootstrap using the installer this seems easier than using MCO, but using MCO for this covers more installation types. This are needs more investigation/work.